src/Pos/Wallet/KeyStorage.hs

```diff
@@ -168,7 +172,7 @@ instance (MonadIO m) => MonadKeys (KeyStorage m) where
     getPrimaryKey = use usPrimKey
     getSecretKeys = use usKeys
     addSecretKey sk =
-        whenM (not . elem sk <$> use usKeys) $
+        whenM (not . (`containsKey` sk) <$> use usKeys) $
```

Comment:
Most likely using `find` from `universum` instead of `containsKey` and `whenJust` would be better.